modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionsSingleMessage.java

```diff
@@ -356,6 +394,9 @@ public void exchangeStartTime(long exchangeStartTime) {
             if (partHistCntrs != null && partHistCntrsBytes == null)
                 partHistCntrsBytes0 = U.marshal(ctx, partHistCntrs);
 
+            if (partClearCntrs != null && partClearCntrsBytes == null)
+                partClearCntrsBytes0 = U.marshal(ctx, partHistCntrs);
```

Comment:
Copy-paste, should be U.marshal(ctx, partClearCntrs);
Need to check why this has not been detected in tests: essentially, partsHistCntr was always used instead of partClearCntrs.

Reply:
Fixed and added a reproducer `testConsistencyOnCounterTriggeredRebalanceClearTombstonesNotCrd`.
Existing tests doesn't reveal the problem because coordinator is a supplier.